pallets/loans-ref/src/lib.rs

```diff
@@ -687,17 +716,26 @@ pub mod pallet {
 			Ok(())
 		}
 
-		fn portfolio_valuation_for_pool(pool_id: PoolIdOf<T>) -> Result<T::Balance, DispatchError> {
+		fn portfolio_valuation_for_pool(
+			pool_id: PoolIdOf<T>,
+		) -> Result<(T::Balance, u32), DispatchError> {
 			let rates = T::InterestAccrual::rates();
-			ActiveLoans::<T>::get(pool_id).into_iter().try_fold(
+			let loans = ActiveLoans::<T>::get(pool_id);
+			let count = loans.len().ensure_into()?;
+			let value = loans.into_iter().try_fold(
 				T::Balance::zero(),
 				|sum, (loan, _)| -> Result<T::Balance, DispatchError> {
 					Ok(sum.ensure_add(loan.current_present_value(&rates)?)?)
 				},
-			)
+			)?;
+
+			Ok((value, count))
```

Comment:
We could (and maybe should?) return the count of rates here to satisfy the `m`, rather than always assuming the max. But 1000 weights is the equivalent of only about three DB reads, so it's not a _huge_ waste of block time to just assume the max

Reply:
Ok, I see it like you. Knowing the values, I would prefer not to give the loans all that knowledge and assume always the maximum